src/dev/flang/ast/Match.java

```diff
@@ -241,6 +241,9 @@ Match assignToField(Resolution res, AbstractFeature outer, Feature r)
    */
   public Expr propagateExpectedType(Resolution res, AbstractFeature outer, AbstractType t)
   {
+    if (CHECKS) check
+      (!t.isVoid() || type().isVoid());
```

Comment:
I fear this might introduce check failures if you do things like
```
  a(v void) =>
  b(a Any) =>
  v choice unit := unit
  a (match v
       unit => panic "bla")
  a (match v
       unit => 4711)
  b (match v
       unit => panic "bla")
  b (match v
       unit => 4711)
```